lib/rspec/matchers/built_in/contain_exactly.rb

```diff
@@ -72,7 +85,10 @@ def message_line(prefix, collection, surface_descriptions=false)
 
         def match(_expected, _actual)
           return false unless convert_actual_to_an_array
-          match_when_sorted? || (extra_items.empty? && missing_items.empty?)
+          matched_when_sorted = match_when_sorted?
+          return true if matched_when_sorted
+          return matched_when_sorted if @transitive
```

Comment:
I'm struggling to understand how this works.
From my understanding, it would clamp here, and would return `false` early before calculating `extra_items` and `missing_items` if `@transitive` is `true`. Is this how it works?

Reply:
However, `extra_items` and `missing_items` will be called later when `failure_message` is called, wouldn't they?